sled-agent/src/rack_setup/service.rs

```diff
@@ -488,10 +516,37 @@ impl ServiceInner {
                     "Initialized sled agent on sled with bootstrap address: {}",
                     bootstrap_addr
                 );
+                Ok(())
+            },
+        ))
+        .await
+        .into_iter()
+        .collect::<Result<_, SetupServiceError>>()?;
 
-                // Next, initialize any datasets on sleds that need it.
+        // Set up internal DNS services.
+        futures::future::join_all(plan.iter().map(
```

Comment:
I'm having trouble understanding exactly how many sleds this request to setup DNS is sent to. I was under the impression there was currently 1 sled selected to run the DNS server, from constant `common::address::DNS_REDUNDANCY`. But this seems to iterate over all elements of the plan, which I thought included every sled to be created. Could you explain this a bit more?

Reply:
The value of `allocation.services_request.dns_services` is getting set when creating the "RSS plan", and is only enacted here. There, the value of `DNS_REDUNDANCY` is taken into consideration. 

Here, we only actually initialize a DNS service if `allocation.services_request.dns_services` is not empty - I've added an explicit check for that to make it more obvious.

Reply:
Thanks for the explanation. The extra filter check there is very helpful, thanks!